@here/harp-mapview/lib/ScreenCollisions.ts

```diff
@@ -15,6 +18,22 @@ const RBush = require("rbush");
 
 const logger = LoggerManager.instance.create("ScreenCollissions");
 
+export interface IBox {
+    minX: number;
+    minY: number;
+    maxX: number;
+    maxY: number;
+    type: string;
```

Comment:
remove type from IBox and in addition have:
```
interface CollisionBox extends IBox {
    type: "box";
}
```
and also replace the type in `LineWithBounds` with `type: "line"`

This gives you some additional type safety.

Reply:
Thanks, thats what I had in mind originally, just didn't come out quite right.